src/config.rs

```diff
@@ -567,10 +567,37 @@ impl Cfg {
                 return Ok(Some((name.into(), reason)));
             }
 
-            // Then look for 'rust-toolchain'
-            let toolchain_file = d.join("rust-toolchain");
-            if let Ok(contents) = utils::read_file("toolchain file", &toolchain_file) {
-                let override_file = Cfg::parse_override_file(contents)?;
+            // Then look for 'rust-toolchain' or 'rust-toolchain.toml'
+            let path_rust_toolchain = d.join("rust-toolchain");
+            let path_rust_toolchain_toml = d.join("rust-toolchain.toml");
+
+            let (toolchain_file, contents, toml_only) = match (
+                utils::read_file("toolchain file", &path_rust_toolchain),
+                utils::read_file("toolchain file", &path_rust_toolchain_toml),
+            ) {
+                (contents, Err(_)) => {
+                    // no `rust-toolchain.toml` exists
+                    (path_rust_toolchain, contents, false)
+                }
+                (Err(_), Ok(contents)) => {
+                    // only `rust-toolchain.toml` exists
+                    (path_rust_toolchain_toml, Ok(contents), true)
+                }
+                (Ok(contents), Ok(_)) => {
+                    // both `rust-toolchain` and `rust-toolchain.toml` exist
+
+                    // TODO: is there a better way to print a warning?
+                    eprintln!(
+                        "warning: both `{0}` and `{1}` exist. Using `{0}`",
+                        path_rust_toolchain.display(),
+                        path_rust_toolchain_toml.display(),
+                    );
```

Comment:
This message should be a notification sent via the notification handler which is in the `notify` local variable set at the top of this function.

Reply:
✓